static/app/components/organizations/projectPageFilter.tsx

```diff
@@ -0,0 +1,493 @@
+      );
+
+      // "All Projects"
+      if (!val.length) {
+        return [ALL_ACCESS_PROJECTS];
+      }
+
+      // "My Projects"
+      if (
+        showNonMemberProjects &&
+        memberProjectsSelected &&
+        val.length === memberProjects.length
+      ) {
+        return [];
+      }
+
+      return val;
+    },
+    [memberProjects, showNonMemberProjects]
+  );
+
+  /**
+   * Transforms an array of page filter URL parameters into a plain array of project
+   * IDs. This is necessary because some URL values denote special configurations: [] =
+   * "My Projects" and [-1] = "All Projects".
+   */
+  const mapURLValueToNormalValue = useCallback(
+    (val: number[]) => {
+      // "All Projects"
+      if (val.includes(ALL_ACCESS_PROJECTS)) {
+        return [];
+      }
+
+      // "My Projects"
+      if (!val.length) {
+        return memberProjects.map(p => parseInt(p.id, 10));
+      }
+
+      return val;
+    },
+    [memberProjects]
+  );
+
+  const [value, setValue] = useState<number[]>(mapURLValueToNormalValue(pageFilterValue));
+
+  const handleChange = useCallback(
+    async (newValue: number[]) => {
+      onChange?.(newValue);
+      setValue(newValue);
+
+      trackAdvancedAnalyticsEvent('projectselector.update', {
+        count: newValue.length,
+        path: getRouteStringFromRoutes(routes),
+        organization,
+        multi: allowMultiple,
+      });
+
+      // Wait for the menu to close before calling onChange
+      await new Promise(resolve => setTimeout(resolve, 0));
+
+      updateProjects(mapNormalValueToURLValue(newValue), router, {
+        save: true,
+        resetParams: resetParamsOnChange,
+        environments: [], // Clear environments when switching projects
+      });
+    },
+    [
+      resetParamsOnChange,
+      router,
+      allowMultiple,
+      organization,
+      routes,
+      onChange,
+      mapNormalValueToURLValue,
+    ]
+  );
+
+  const onToggle = useCallback(
+    newValue => {
+      trackAdvancedAnalyticsEvent('projectselector.toggle', {
+        action: newValue.length > value.length ? 'added' : 'removed',
+        path: getRouteStringFromRoutes(routes),
+        organization,
+      });
+    },
+    [value, routes, organization]
+  );
+
+  const onReplace = useCallback(() => {
+    trackAdvancedAnalyticsEvent('projectselector.direct_selection', {
+      path: getRouteStringFromRoutes(routes),
+      organization,
+    });
+  }, [routes, organization]);
+
+  const onClear = useCallback(() => {
+    trackAdvancedAnalyticsEvent('projectselector.clear', {
+      path: getRouteStringFromRoutes(routes),
+      organization,
+    });
+  }, [routes, organization]);
+
+  const options = useMemo<SelectOptionOrSection<number>[]>(() => {
+    const hasProjects = !!memberProjects.length || !!nonMemberProjects.length;
+    if (!hasProjects) {
+      return [];
+    }
+
+    const getProjectItem = (project: Project) => {
+      return {
+        value: parseInt(project.id, 10),
```

Comment:
Do we need to handle the `NaN` case here?

Reply:
Hmm I'm not sure… Currently we don't do it since `project.id` will always be a numeric string.
https://github.com/getsentry/sentry/blob/70a431d4b09a90d600477d4939a71637380ff682/static/app/components/organizations/projectSelector/index.tsx#L206-L208

Reply:
Yeah that's fine, just wanted to make sure!